exp/operator/controllers/genericprovider_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	operatorv1 "sigs.k8s.io/cluster-api/exp/operator/api/v1alpha4"
+	"sigs.k8s.io/cluster-api/util/conditions"
+)
+
+func TestConvertStructuredProviderToGeneric(t *testing.T) {
+	g := NewWithT(t)
+
+	coreProvider := &operatorv1.CoreProvider{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "core",
+			Namespace: "namespace",
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "CoreProvider",
+			APIVersion: "operator.cluster.x-k8s.io/v1alpha4",
+		},
+		Status: operatorv1.CoreProviderStatus{
+			ProviderStatus: operatorv1.ProviderStatus{
+				Conditions: clusterv1.Conditions{
+					*conditions.TrueCondition(operatorv1.PreflightCheckCondition),
+				},
+			},
+		},
+	}
+
+	genericProvider, err := convertStructuredProviderToGeneric(coreProvider)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(reflect.DeepEqual(genericProvider.ObjectMeta, coreProvider.ObjectMeta)).To(BeTrue())
+	g.Expect(reflect.DeepEqual(genericProvider.TypeMeta, coreProvider.TypeMeta)).To(BeTrue())
+	g.Expect(reflect.DeepEqual(genericProvider.Status.Conditions, coreProvider.Status.Conditions)).To(BeTrue())
```

Comment:
Do we not need to check the Spec and entire Status? (This applies to other tests too)